docs/cd_with_aws.md

```diff
@@ -0,0 +1,46 @@
+# Continuous Deployment with AWS
+
+This document explains how to set up Continuous Deployment (CD) with AWS using GitHub Actions.
+
+## Prerequisites
+
+Before you start, make sure you have the following:
+
+1. **AWS Account**: You need an AWS account. Sign up [here](https://aws.amazon.com/).
+
+2. **Amazon ECR (Elastic Container Registry) Setup**: 
+   - Create a new repository in Amazon ECR.
+   - Note down the repository URI, which will be used in the GitHub Actions workflow.
+
+3. **AWS Credentials**: 
+   - AWS Access Key ID
+   - AWS Secret Access Key
+   - These credentials should have permission to interact with ECR and ECS.
+
+4. **Create Environments**:
+
+   The GitHub Actions workflow will automatically deploy to the correct environment based on the branch being pushed to. The branch `main` will always be linked to the `production` environment, while other branches will use their own names as the environment. All environments added in GitHub must have the same name as the branches.
+
+5. **GitHub Repository Setup**:
+   - Secrets: Add the following secrets to your GitHub repository:
```

Comment:
These secrets are environment specific right? Maybe we should clarify that these are not repo level settings, but environment level